rustfmt-core/src/reorder.rs

```diff
@@ -233,7 +233,7 @@ impl ReorderableItemKind {
         match item.node {
             _ if contains_macro_use_attr(item) => ReorderableItemKind::Other,
             ast::ItemKind::ExternCrate(..) => ReorderableItemKind::ExternCrate,
-            ast::ItemKind::Mod(..) => ReorderableItemKind::Mod,
+            ast::ItemKind::Mod(ref m) if m.inner.hi() != item.span.hi() => ReorderableItemKind::Mod,
```

Comment:
Could you either factor out the test into a function, or add a comment please (it's a bit unclear what the guard is checking for).

Reply:
There is `is_mod_decl` in items.rs, so I am going to use that.